src/templates/speciality.js

```diff
@@ -65,22 +708,31 @@ const Speciality = ({
   const books = getExternalType(speciality, `Book`)
```

Comment:
As we're already doing a big transformation, you could probably move all of these methods to be included in the `flattenSpeciality` method. That way you can just pass `flattenedSpeciality` to the components

Reply:
actually, I'd argue against it because then you'd still need to pass either `flattenedSpeciality.talks` OR you'd need to destructure the whole object and why do that when only one key is needed?

Reply:
Indeed but what I'm getting at is that we're mixing where our data is coming from. 

`flattenedSpeciality` should become the main source of all of the data, here is where we're going a lot preparation to pass the data in to the components. So by leaving the `getExternalType` method to extract data from the query, you bypass the transformation and thus reduce the reliability of the data created.

`why do that when only one key is needed` 
It doesn't matter, we should follow our own rules we've decided and maintain consistency.

Reply:
`props -> transformation -> transformedProps -> data passed to  components` 

compared to 
`props -> transformation -> transformedProps -> data passed to  components` 
plus
`props -> extractionMethod -> data passed to components`

Reply:
The reason for `flattenSpeciality` is to generate data in a shape that is flatten and assign defaults but most importantly something we can trust right?

For example - what happens if we request `rxjs` speciality but it doesn't have any externalResources returned from the query? [it breaks](http://localhost:8000/___graphql?query=%7B%0A%20%20contentfulSpeciality(slug%3A%20%7Beq%3A%20%22rxjs%22%7D)%20%7B%0A%20%20%20%20slug%0A%20%20%20%20externalResources%20%7B%0A%20%20%20%20%20%20id%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A&operationName=undefined), because we're trying to filter `speciality.externalResources`, which as you can see from the query response is `null`. What would happen if we created `externalResources` on `flattenedSpeciality` like this `Get('externalResources', speciality, [])`, we'd be ok and talks, books and projects wouldn't render successfully.

The main aim of this PR is to handle these types of issues, tying up potential bugs so that contentful can give us whatever it wants and we don't have broken pages or deployment errors.

Reply:
Ok, I pushed a thing which I think matches your expectations.